solid/src/main/java/com/inrupt/client/solid/SolidClient.java

```diff
@@ -131,36 +135,37 @@ public <T extends Resource> CompletionStage<T> read(final URI identifier, final
         headers.firstValue(USER_AGENT).ifPresent(agent -> builder.setHeader(USER_AGENT, agent));
 
         final Request request = builder.build();
-        return client.send(request, Response.BodyHandlers.ofByteArray())
```

Comment:
This looks like a large change, but essentially the error branch of the initial `if` has been moved to the wrapping `throwOnError`